### PR TITLE
#219 - Fix MavenProxySlice response

### LINKED ISSUE #219: MavenProxySlice should return 404 when artifact isn't got
Maven proxy should return `404` when artifact wasn't found in cache and wasn't got via proxy.
After fixing disable test.
See #218

### PR DESCRIPTION
**Pull request by genryxy**
Closes #219 
Fix `MavenProxySlice` response when not found

### CODE REVIEW

File: src/main/java/com/artipie/maven/http/CachedProxySlice.java [hunk 1]
```diff
@@ -123,7 +126,18 @@ public Response response(final String line, final Iterable<Map.Entry<String, Str
...
+                            if (pub.size().isPresent()) {
+                                resp = new AsyncResponse(new PublisherAs(pub)
+                                    .bytes()
+                                    .thenApply(CachedProxySlice::contentContainsNotFound)
+                                );
+                            } else {
+                                resp = new RsWithBody(StandardRs.OK, new Content.From(pub));
+                            }
+                            return resp;
+                        }
```
Comment (olegmoz) [inline src/main/java/com/artipie/maven/http/CachedProxySlice.java:140]: @genryxy the problem with this code right now is that you are reading whole content for every request into memory, including `.jar` files which might be quite large. It is possible to handle response status and not read content at all, as it is done in PyPi adapter: https://github.com/artipie/pypi-adapter/blob/master/src/main/java/com/artipie/pypi/http/ProxySlice.java

**Review by olegmoz — CHANGES_REQUESTED** [review]

**Comment by genryxy** [comment]
@olegmoz thanks, added puzzle to change implementation. I tried the similar approach as in pypi-adapter, but `MavenProxySliceAuthIT#shouldGet` hangs. Probably it happens because cache in pypi-adapter is wrapped in `FromRemoteCache` or maybe there is problem with requests. I was not able to identify the source of problem.

**Review by olegmoz — APPROVED** [review]
@genryxy thanks, merging

**Comment by 0crat** [comment]
Job `gh:artipie/maven-adapter#220` is not assigned, can't get performer

**Pull request merged**